Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -65,6 +65,10 @@ enum SOResultCode : s32
 NetIPTopDevice::NetIPTopDevice(Kernel& ios, const std::string& device_name)
     : Device(ios, device_name)
 {
+  m_queue.Reset([this](AsyncTask task) {
+    const IPCReply reply = task.handler();
+    m_async_replies.emplace(AsyncReply{task.request, reply.return_value});
```

Comment:
This does not seem thread safe. This callback will be called from the WorkQueueThread on a separate thread and NetIPTopDevice::Update might be reading from the same queue (m_async_replies) at the same time.

Reply:
Good catch. I added lock guards to prevent concurrent access to `m_async_replies`.